src/equations/compressible_euler_1d.jl

```diff
@@ -390,16 +399,17 @@ positive axis direction), determined by the argument `which` set to
   lambda2 = v1 + a
   lambda3 = v1 - a
 
-  lambda1_p = 0.5 * (lambda1 + abs(lambda1))
-  lambda2_p = 0.5 * (lambda2 + abs(lambda2))
-  lambda3_p = 0.5 * (lambda3 + abs(lambda3))
+  lambda1_p = positive_part(lambda1)
+  lambda2_p = positive_part(lambda2)
+  lambda3_p = positive_part(lambda3)
```

Comment:
On my system at home with a single thread, this change improved the performance of the Taylor-Green vortex setup by ca. 10%. 

> It makes the formulas less obvious when compared to those on paper. 

I guess this depends on where you get the formulas on paper from. Some authors prefer to write something like $\lambda_1^+$ to refer to the positive part instead of $(\lambda_1 + |\lambda_1|) / 2$.

> I always thought that logic was more expensive that a built-in function.

That depends on the case. Here, we use `ifelse`, so that both branches are always evaluated, allowing SIMD optimizations (if LLVM is capable of doing so). Let me run some benchmarks on another system again to be sure.

Reply:
On my system at work, it's a bit less pronounced but still in the realm of 7% improvement of the PID for the Taylor-Green vortex with setup
```julia
trixi_include("examples/tree_3d_fdsbp/elixir_euler_taylor_green_vortex.jl", tspan=(0.0, 1.0), analysis_interval=1000)
```

Reply:
Okay, I didn't realize that the `ifelse` evaluates both branches in this way to allow SIMD to hide this cost. I suppose this "avoid logic" is just a holdover from my Fortran brain :).

For the formulas maybe we can put a small comment above, e.g., `positive_part` with "# Compute $(\lambda + |\lambda|) / 2$" that way we get the best of both worlds, the performance increase and then new users don't need to lookup the new function in the `math.jl`.

Reply:
Sounds reasonable - done in 6d89a67a

Note that it's not that hard in a modern editor like VScode, providing the docstring while hoovering over the function name:
![Screenshot from 2022-11-30 12-03-23](https://user-images.githubusercontent.com/12693098/204780499-deb38b2d-9743-4a5d-9b25-8c572177ee6f.png)

Reply:
This should resolve all of the issues, doesn't it, @andrewwinters5000? Thanks for your review :slightly_smiling_face:

Reply:
Yup, that looks good! I will approve.